tests/test_astrometryTask.py

```diff
@@ -203,25 +203,43 @@ def doTest(self, pixelsToTanPixels, order=3):
         )
         self.assertLess(len(resultsNoFitRefSelect.matches), len(resultsNoFit.matches))
 
-    def makeSourceCat(self, distortedWcs):
-        """Make a source catalog by reading the position reference stars and distorting the positions
+    def makeSourceCat(self, wcs, sourceSchema=None, doScatterCentroids=False):
+        """Make a source catalog by reading the position reference stars using
+        the proviced WCS.
+
+        Optionally provide a schema for the source catalog (to allow
+        AstrometryTask in the test methods to update it with the
+        "calib_astrometry_used" flag).  Otherwise, a minimal SourceTable
+        schema will be created.
+
+        Optionally, via doScatterCentroids, add some scatter to the centroids
+        assiged to the source catalog (otherwise they will be identical to
+        those of the reference catalog).
         """
-        loadRes = self.refObjLoader.loadPixelBox(bbox=self.bbox, wcs=distortedWcs, filterName="r")
+        loadRes = self.refObjLoader.loadPixelBox(bbox=self.bbox, wcs=wcs, filterName="r")
         refCat = loadRes.refCat
         refCentroidKey = afwTable.Point2DKey(refCat.schema["centroid"])
         refFluxRKey = refCat.schema["r_flux"].asKey()
 
-        sourceSchema = afwTable.SourceTable.makeMinimalSchema()
-        measBase.SingleFrameMeasurementTask(schema=sourceSchema)  # expand the schema
+        if sourceSchema is None:
+            sourceSchema = afwTable.SourceTable.makeMinimalSchema()
+            measBase.SingleFrameMeasurementTask(schema=sourceSchema)  # expand the schema
         sourceCat = afwTable.SourceCatalog(sourceSchema)
         sourceCentroidKey = afwTable.Point2DKey(sourceSchema["slot_Centroid"])
         sourceInstFluxKey = sourceSchema["slot_ApFlux_instFlux"].asKey()
         sourceInstFluxErrKey = sourceSchema["slot_ApFlux_instFluxErr"].asKey()
 
+        scatterFactor = 1.0
+        np.random.seed(12345)
+
         sourceCat.reserve(len(refCat))
         for refObj in refCat:
+            if doScatterCentroids:  # add some small random offsets to source centroids
+                scatterFactor = np.random.uniform(0.999, 1.001)
```

Comment:
This is totally irrelevant to the ticket, but seeing this loop over sources instead of doing arrays is making me a bit 🤢.  No need to change, but now I'm on record.

Reply:
Well, if only your gag-reflex kicked in when you were last updating this file 😜. I gave it my best effort (I got a bit hung up on setting the centroids...), so now you get to review my changes to make sure I didn't mess it up (the tests do pass, and I did check that mine failed in the same manner with the un-updated `with self.assertRaisesRegex`, so hopefully all is ok)!